### ISSUE TITLE
Quantile broken if only one centroid

### ISSUE THREAD

**Issue by pjz** [CamDavidsonPilon/tdigest#17]
```
>>> from tdigest import TDigest as TD
>>> td = TD()
>>> td.update(1)
>>> td.quantile(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/tdigest/tdigest.py", line 184, in quantile
    delta = (c_i.mean - self.C.prev_item(key)[1].mean) / 2.
  File "/usr/local/lib/python2.7/dist-packages/bintrees/abctree.py", line 684, in prev_item
    raise KeyError(str(key))
KeyError: '1.0'
```

a little bit of inspection shows that `quantile` breaks if there's only one Centroid in `self.C`.

**Comment by CamDavidsonPilon** [comment]
Thanks for reporting this. I'll fix this shortly, unless you have PR in mind

**Comment by pjz** [comment]
I have some tests, but they're fairly trivial.  I have a PR, but I'm not entirely sure the math is correct on it.  I'll submit shortly and you can have a look.

**Comment by pjz** [comment]
Looking forward to a fix, even if it's not mine!

**Comment by pjz** [comment]
Fixed by #19
